src/os/bluestore/bluefs_types.h

```diff
@@ -98,6 +133,10 @@ struct bluefs_fnode_t {
       extents.push_back(ext);
     }
     allocated += ext.length;
+
+    // update new extents also - for simplicity skip coalescing extents
+    delta.extents.push_back(ext);
```

Comment:
IMO one should update other methods altering extents collection then. Namely pop_front_extent, swap_extents and clear_extents. Just to make sure fnode is always in a consistent state.

Reply:
and honestly the idea to keep just the starting position for delta and build the delta list itself on demand (i.e. prior to calling op_file_update_inc) looks more attractive to me at this point

Reply:
Added calculation of fnode_delta_t (code mostly copied from yours) in make_delta.